stage0/src/sev.rs

```diff
@@ -26,16 +27,108 @@ use oak_sev_guest::{
     },
 };
 use spinning_top::Spinlock;
+use strum::FromRepr;
 use x86_64::{
     instructions::tlb,
-    registers::control::Cr3,
+    registers::{control::Cr3, model_specific::Msr},
     structures::paging::{
         frame::PhysFrameRange, page::AddressNotAligned, Page, PageSize, PageTable, PageTableFlags,
         PhysFrame, Size1GiB, Size2MiB, Size4KiB,
     },
     PhysAddr, VirtAddr,
 };
 
+/// The cache memory type used with MTRR.  We only use Write-Protect mode which the Linux kernel
+/// expects to be enabled by the firmware in order to enable SEV.
+#[repr(u8)]
+#[allow(dead_code)] // Remove if this is ever ported to a public crate.
+#[derive(FromRepr)]
+pub enum MemoryType {
+    UC = 0, // Uncacheable.
+    WC = 1, // Write-Combining.
+    WT = 4, // Writethrough	Reads.
+    WP = 5, // Write-Protect.
+    WB = 6, // Writeback.
+}
+
+impl TryFrom<u8> for MemoryType {
+    type Error = &'static str;
+    fn try_from(value: u8) -> Result<MemoryType, &'static str> {
+        match MemoryType::from_repr(value) {
```

Comment:
The idiomatic solution is to use `Option::ok_or()` here instead of the pattern match.

Reply:
Oops... I missed this.  I'll take care of it now.